src/ResourceManager/Network/Commands.Network/ApplicationGateway/AuthenticationCertificate/AddAzureApplicationGatewayAuthenticationCertificateCommand.cs

```diff
@@ -0,0 +1,48 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using System;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Add, "AzureRmApplicationGatewayAuthenticationCertificate"), OutputType(typeof(PSApplicationGateway))]
```

Comment:
These are child resources of application gateway. I don't see SupportsShouldProcess  being added to any other child resources (e.g. AddAzureLoadBalancerBackendAddressPoolConfigCommand.cs for load balancer child resource). Also VS says it cannot resolve SupportsShouldProcess  in this file.